### DISCUSSION TITLE
What Python versions are supported?

### DISCUSSION THREAD

**Discussion by anton-m-py** [strawberry-graphql/strawberry#3294] [Q&A] [answered]
Specifically, I would like to know if Python 3.12 is supported. I encountered a strange bug with latest strawberry on Python 3.12 and am curious to know if it is not supported yet.

**Comment by patrick91** [comment, marked as answer]
hi @bhumkong, we support from 3.8 to 3.12 (we are testing all these versions in our test suite) 😊

what issue are you getting?

**Comment by anton-m-py** [comment]
The error was `'Schema' object has no attribute 'query_type'` when I created schema like that:
```
from strawberry import type, field, Schema, mutation


@type
class Query:
    @field
    def hello(self, name: str) -> str:
        return f"Hello {name}!"


schema = Schema(Query)
```
But now I tried running it again on Python 3.12 and I can't reproduce it.

**Reply by patrick91** [reply]
oh, odd, let me know if it happens again 😊
